pkg/chunk/aws/s3_storage_client.go

```diff
@@ -38,8 +43,24 @@ func init() {
 // S3Config specifies config for storing chunks on AWS S3.
 type S3Config struct {
 	S3               flagext.URLValue
-	BucketNames      string
 	S3ForcePathStyle bool
+
+	BucketNames     string
+	Endpoint        string             `yaml:"endpoint"`
+	Region          string             `yaml:"region"`
+	AccessKeyID     string             `yaml:"access_key_id"`
+	SecretAccessKey string             `yaml:"secret_access_key"`
+	Insecure        bool               `yaml:"insecure"`
+	SSEEncryption   bool               `yaml:"sse_encryption"`
+	PutUserMetadata map[string]*string `yaml:"put_user_metadata"`
```

Comment:
How is this intended to be used?

Reply:
I'm going to remove this option.  It can be added later if we want it.  It was still on my list to test how yaml would unmarshal to `map[string]*string`.  My current answer is:  I don't know :).

I'd rather get this PR in and not worry about this field, then add appropriate testing around this.